### ISSUE TITLE
Restore-DbaDatabase with -OutputScriptOnly sets database to RESTRICTED_USER

### ISSUE THREAD

**Issue by backintheday41** [sqlcollaborative/dbatools#2940]
### General Troubleshooting steps

- [x] Verified running the latest release of dbatools?
- [x] Verified errors are not related to permissions?
- [x] Can duplicate in new/clean PowerShell session (clean = `powershell -NoProfile`)?

### Version Information
 - Operating System (Name|Version):  Support package attached.
 - PowerShell Version:  Support package attached.
 - SQL Server (Edition|Version):  Support package attached.

## Steps to Reproduce
Support package attached.
 - [ ] Attach any screenshots (if possible/allowed)
 - [ ] Attach logs generated by dbatools (`Get-DbatoolsLog`)
 - [ ] Attach output from PowerShell console (if possible/allowed)

## Feature Request
When using the -OutputScriptOnly option, the targeted database should not be put into RESTRICTED_USER mode.

### Problem to solve
Same as above.

### Additional information
[dbatools_support_pack_2017_12_28-11_49_11.zip](https://github.com/sqlcollaborative/dbatools/files/1591663/dbatools_support_pack_2017_12_28-11_49_11.zip)

**Comment by potatoqualitee** [comment]
oh fk, I can repro. Will get the whole team on this immediately.

**Comment by potatoqualitee** [comment]
found it in Invoke-DbaAdvancedRestore.ps1
```
C:\github\dbatools\functions\Invoke-DbaAdvancedRestore.ps1:154:                            $null = $server.Query("Alter database $Database set offline with rollback
immediate; alter database $Database set restricted_user; Alter database $Database set online with rollback immediate",'master')
```

**Comment by potatoqualitee** [comment]
We should have this resolved within the hour

**Comment by backintheday41** [comment]
Wow, that was fast!  Thanks!

**Comment by backintheday41** [comment]
It is still happening with version 0.9.141.  I reported the with version 0.9.140.  I updated dbatools (with Update-Dbatools).  Should I expect the fix in a later release?  Would you like another SupportPackage against 0.9.141?

**Comment by niphlod** [comment]
lemme check real quick

**Comment by niphlod** [comment]
damn, confirmed.

**Comment by potatoqualitee** [comment]
damn wth, k we'll get it fixed and will do a regression test for tomorrow

**Comment by potatoqualitee** [comment]
We have now addressed the issue, most definitely in 0.9.142. Sorry for the false start! In addition to manual testing, @niphlod also created a test that runs with each commit to ensure that this does not happen again. Our apologies for the inconvenience.

**Comment by backintheday41** [comment]
Confirmed that we're good on 0.9.142. Thank you for your quick follow-up and getting right on it it.  Much appreciated!

Have a safe and happy New Year :)
